synology_dsm/api/core/upgrade.py

```diff
@@ -0,0 +1,23 @@
+"""DSM Upgrade data and actions."""
+
+
+class SynoCoreUpgrade:
+    """Class containing upgrade data and actions."""
+
+    API_KEY = "SYNO.Core.Upgrade"
+    API_SERVER_KEY = API_KEY + ".Server"
+
+    def __init__(self, dsm):
+        self._dsm = dsm
+        self._data = {}
+
+    def update(self):
+        """Updates Upgrade data."""
+        raw_data = self._dsm.get(self.API_SERVER_KEY, "check")
+        if raw_data:
+            self._data.update(raw_data["data"])

```

Comment:
```suggestion
            self._data = raw_data["data"]
```

I don't think we should save previous update data